### PR TITLE
SASS-4738: Get Other Employment Details from DES and expose them via an API

### PR DESCRIPTION
**Pull request by vivekhmrc**
### Description
The employment backend gets Employment Other Income data from DES/IF and exposes it via an API

Add a link to the relevant story in Jira
[SASS-4738](https://jira.tools.tax.service.gov.uk/browse/SASS-4738)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/services/OtherEmploymentIncomeService.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class OtherEmploymentIncomeService @Inject()(otherEmploymentConnector: OtherEmploymentIncomeConnector) {
+
+
+    def getOtherEmploymentIncome(nino: String, taxYear: Int, mtditid: String)
+                                (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[OtherEmploymentIncomeResponse] = {
+    otherIncome(nino, taxYear, mtditid).flatMap {
```
Comment (rtna-hmrc) [inline app/services/OtherEmploymentIncomeService.scala:31]: I am trying to understand this piece of code. We are doing a match case here from a Future, and for each case we return the exact same result. The way I understand this, should we not just do `otherEmploymentConnector.getOtherEmploymentIncome(nino, taxYear)(hc.withExtraHeaders("mtditid" -> mtditid))` and that's it?

File: conf/app.routes [hunk 2]
```diff
@@ -8,4 +8,6 @@ DELETE  /income-tax/nino/:nino/sources/:employmentId/unignore       controllers.
 
 DELETE  /income-tax/nino/:nino/sources/:employmentId/:toRemove      controllers.DeleteEmploymentController.deleteOrIgnoreEmployment(nino: String, employmentId: String, toRemove: String, taxYear: Int)
 
-POST    /income-tax/nino/:nino/sources                              controllers.CreateUpdateEmploymentController.createUpdateEmployment(nino: String, taxYear: Int)
\ No newline at end of file
+POST    /income-tax/nino/:nino/sources                              controllers.CreateUpdateEmploymentController.createUpdateEmployment(nino: String, taxYear: Int)
+
+GET     /income-tax/nino/:nino/other/employment/income              controllers.OtherEmploymentIncomeController.getOtherEmploymentIncome(nino: String, taxYear: Int)
```
Comment (rtna-hmrc) [inline conf/app.routes:13]: the backend APIs have `income/other/employments`. Any reason to not leave it the same way? Not that any convinces me, but I am leaning towards keeping it consistent

File: test/connectors/parsers/OtherEmploymentIncomeHttpParserSpec.scala [hunk 3]
```diff
@@ -20,60 +20,66 @@ import connectors.errors.{ApiError, SingleErrorBody}
...
-class OtherEmploymentsIncomeHttpParserSpec extends UnitTest {
+class OtherEmploymentIncomeHttpParserSpec extends UnitTest {
 
   private val anyHeaders: Map[String, Seq[String]] = Map.empty
   private val anyMethod: String = "GET"
   private val anyUrl = "/any-url"
   private val singleErrorBody: SingleErrorBody = SingleErrorBody("PARSING_ERROR", "Error parsing response from DES")
   private val singleErrorBodyJson: JsValue = Json.toJson(singleErrorBody)
-  private val underTest = OtherEmploymentsIncomeHttpParser.OtherEmploymentsIncomeHttpReads
+  private val employmentOtherIncomeHttpReads = OtherEmploymentIncomeHttpParser.OtherEmploymentIncomeHttpReads
```
Comment (rtna-hmrc) [inline test/connectors/parsers/OtherEmploymentIncomeHttpParserSpec.scala:33]: We use to have the `underTest` pattern everywhere, I'd keep it like the same way for now if possible? We could have a group discussion about it

File: app/connectors/OtherEmploymentIncomeConnector.scala [hunk 4]
```diff
@@ -17,24 +17,24 @@
...
 import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
 import utils.DESTaxYearHelper.desTaxYearConverter
 
 import java.net.URL
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class GetOtherEmploymentIncomeConnector @Inject()(val http: HttpClient,
-                                                  val appConfig: AppConfig)(implicit ec: ExecutionContext) extends DesConnector {
+class OtherEmploymentIncomeConnector @Inject()(val http: HttpClient,
```
Comment (rtna-hmrc) [inline app/connectors/OtherEmploymentIncomeConnector.scala:28]: Why remove the `Get`? I am thinking now if we should remove the Get/create and add the methods inside this same connector

File: test/services/OtherEmploymentIncomeServiceSpec.scala [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      val desApiError = ApiError(SERVICE_UNAVAILABLE, serviceUnavailableErrorModel)
+
+      (mockOtherEmploymentIncomeConnector.getOtherEmploymentIncome(_: String, _: Int)(_: HeaderCarrier))
+        .expects(*, *, *)
+        .returning(Future.successful(Left(desApiError)))
+
+      val result = otherEmploymentIncomeService.getOtherEmploymentIncome(nino, taxYear, mtdItID)
+      await(result) mustBe Left(desApiError)
+    }
+
```
Comment (rtna-hmrc) [inline test/services/OtherEmploymentIncomeServiceSpec.scala:68]: minor comment, we can remove this empty lines?

File: conf/app.routes (hunk 2 above)
Comment (vivekhmrc) [reply conf/app.routes:13]: I agree. We can keep it the same as it is in DES. I will incorporate it

File: test/connectors/parsers/OtherEmploymentIncomeHttpParserSpec.scala (hunk 3 above)
Comment (vivekhmrc) [reply test/connectors/parsers/OtherEmploymentIncomeHttpParserSpec.scala:33]: Sure. I'll keep it the same for consistency and then we can refactor it later for all as a separate Tech Debt.

File: test/services/OtherEmploymentIncomeServiceSpec.scala (hunk 5 above)
Comment (vivekhmrc) [reply test/services/OtherEmploymentIncomeServiceSpec.scala:68]: Sure

File: app/services/OtherEmploymentIncomeService.scala (hunk 1 above)
Comment (vivekhmrc) [reply app/services/OtherEmploymentIncomeService.scala:31]: Done

File: app/connectors/OtherEmploymentIncomeConnector.scala (hunk 4 above)
Comment (rtna-hmrc) [reply app/connectors/OtherEmploymentIncomeConnector.scala:28]: tech debt task created here: https://jira.tools.tax.service.gov.uk/browse/SASS-5147

**Pull request merged**